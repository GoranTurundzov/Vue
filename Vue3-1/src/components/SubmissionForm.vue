<template>
   <div class="content content-boxed bg-light">
        <div class="push-left">
            <button 
                class="btn btn-gray push-7-l"
                data-testid="go-back-form-button" 
                @click="closeForm"
            >
                <i class="fas fa-angle-left push-10-r"></i>Back
            </button>
        </div>
        <div class="justify-content-center">
            <h3>{{ title }}</h3>
        </div>
        <div class="content-narrow">
            <form class="form-horisontal block-content content" @submit="submitForm">
                <div class="row">
                    <div class="form-group" :class="{ 'has-error': !validName }">
                        <div class="col-xs-8 ">
                            <label for="Name" class="form-label floating open">Name*</label>
                        </div>
                        <div class="col-xs-12">
                            <input
                                v-model="submission.name"
                                type="text"
                                name="Name"
                                class="form-control"
                                @focus="removeNameError"
                                data-testid="name-input"
                            />
                        </div>
                    </div>
                    <div class="es-required-field col-md-6" v-if="!validName">
                        <i class="fas fa-exclamation-circle"></i>
                        This field is required.
                    </div>
                </div>
                <div class="row">
                    <div class="form-group" :class="{ 'has-error': emailError }">
                        <div class="col-xs-8">
                            <label for="Email" class="form-label floating open">Email*</label>
                        </div>
                        <div class="col-xs-12">
                            <input
                                v-model="submission.email"
                                type="text"
                                name="Email"
                                class="form-control"
                                @focus="removeEmailError"
                                data-testid="email-input"
                            />
                        </div>
                    </div>
                    <div class="es-required-field col-md-6 " v-if="!validEmail">
                        <i class="fas fa-exclamation-circle"></i>
                        This field is required.
                    </div>
                    <div class="es-required-field col-md-6 " v-if="!validEmailFormat">
                        <i class="fas fa-exclamation-circle"></i>
                        Enter a valid e-mail
                    </div>
                </div>
                <div class="row">
                    <div class="form-group">
                        <div class="col-xs-8">
                            <label for="Recommendation" class="form-label floating open">
                                Recommendation
                            </label>
                        </div>
                        <div class="col-xs-12">
                            <select
                                name="Recommendation"
                                v-model="submission.recommendation"
                                class="form-control"
                                data-testid="recommendation-input"
                            >
                                <option value="yes">Yes</option>
                                <option value="no">No</option>
                                <option value="maybe">Maybe</option>
                            </select>
                        </div>
                    </div>
                </div>
                <div class="row">
                    <div class="form-group" :class="{ 'has-error': !validFeedback }">
                        <div class="col-xs-8">
                            <label for="Feedback" class="form-label floating open">Feedback*</label>
                        </div>
                        <div class="col-xs-12">
                            <textarea
                                name="Feedback"
                                v-model="submission.feedback"
                                rows="5"
                                class="form-control"
                                @focus="removeFeedbackError"
                                data-testid="feedback-input"
                            ></textarea>
                        </div>
                    </div>
                    <div class="es-required-field col-md-6" v-if="!validFeedback">
                        <i class="fas fa-exclamation-circle"></i>
                        This field is required.
                    </div>
                </div>
                <div class="row">
                    <div class="form-group" :class="{ 'has-error': dateError }">
                        <div class="col-xs-8">
                            <label for="Date" class="form-label floating open" type="date"
                                >Date*</label
                            >
                        </div>
                        <div class="col-xs-12">
                            <input
                                v-model="submission.date"
                                class="form-control"
                                type="text"
                                name="Date"
                                placeholder="DD/MM/YYYY"
                                @focus="removeDateError"
                                data-testid="date-input"
                            />
                        </div>
                    </div>
                    <div class="es-required-field col-md-6" v-if="dateError">
                        <div v-if="!validDate">
                            <i class="fas fa-exclamation-circle"></i>
                            This field is required.
                        </div>
                        <div v-if="!validDateFormat">
                            <i class="fas fa-exclamation-circle"></i>
                            Enter a valid date in dd/MM/yyyy format only.
                        </div>
                    </div>
                </div>
                <div class="row">
                    <div class="form-group" :class="{ 'has-error': !validForm }">
                        <div class="col-xs-8">
                            <label for="Form" class="form-label floating open">Form*</label>
                        </div>
                        <div class="col-xs-12">
                            <select
                                v-model="submission.form"
                                name="Form"
                                class="form-control"
                                @change="removeFormError"
                                data-testid="form-input"
                            >
                                <option disabled selected value="">Select a form</option>
                                <option
                                    v-for="option of formSelectOptions"
                                    :value="option"
                                    :key="option.name"
                                    >{{ option.name }} - {{ option.form_type }}</option
                                >
                            </select>
                        </div>
                    </div>
                    <div class="es-required-field col-md-6" v-if="!validForm">
                        <i class="fas fa-exclamation-circle"></i>
                        This option is required.
                    </div>
                </div>
                <div class="row">
                    <div class="form-group">
                        <div class="col-xs-8">
                            <label for="Status" class="form-label floating open">Status</label>
                        </div>
                        <div class="col-xs-12">
                            <select
                                name="Status"
                                v-model="submission.status"
                                class="form-control"
                                data-testid="status-input"
                            >
                                <option value="pending">Pending</option>
                                <option value="approved">Approved</option>
                                <option value="rejected">Rejected</option>
                            </select>
                        </div>
                    </div>
                </div>

                <div class="form-group row">
                    <div class="col-xs-12 text-center">
                        <button
                            type="submit"
                            class="btn btn-primary form-material"
                            data-testid="submit-submission-button"
                        >
                            Submit
                        </button>
                    </div>
                </div>
            </form>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        submissionToEdit: {
            type: Object,
            description: 'prop which will be passed to fill out the form with the submission to edit'
        },
        formSelectOptions: {
            type: Array,
            required: true,
            description: 'Options for the form select'
        },
        editedSubmissionIndex: {
            type: Number,
            default: -1,
            description: 'if this is passed the form will be used for editing'
        }
    },
    emits: ['close-form', 'create-new-submission', 'edit-the-submission'],
    data() {
        return {
            submission: {
                name: '',
                email: '',
                recommendation: 'yes',
                feedback: '',
                date: '',
                form: '',
                status: 'pending'
            },
            validName: true,
            validEmail: true,
            validFeedback: true,
            validForm: true,
            validDate: true,
            validDateFormat: true,
            validEmailFormat: true
        };
    },
    computed: {
        dateError() {
            if (!this.validDate || !this.validDateFormat) {
                return true;
            }
            return false;
        },
        emailError() {
            if (!this.validEmail || !this.validEmailFormat) {
                return true;
            }
            return false;
        }
    },
    created() {
        if (this.editedSubmissionIndex >= 0) {
            this.title = 'Edit ' + this.submissionToEdit.name + ' Submission';
            this.submission.name = this.submissionToEdit.name;
            this.submission.email = this.submissionToEdit.email;
            this.submission.recommendation = this.submissionToEdit.recommendation;
            this.submission.feedback = this.submissionToEdit.feedback;
            this.submission.form = {
                name: this.submissionToEdit.form,
                form_type: this.submissionToEdit.form_type
            };
            this.submission.status = this.submissionToEdit.status;
            this.submission.date = this.submissionToEdit.date;
        } else {
            this.title = 'Create new submission';
        }
    },
    methods: {
        submitForm(e) {
            e.preventDefault();
            this.validName = this.submission.name.trim() === '' ? false : true;
            this.validEmail = this.submission.email.trim() === '' ? false : true;
            this.validFeedback = this.submission.feedback.trim() === '' ? false : true;
            this.validForm = this.submission.form === '' ? false : true;
            this.validDate = this.submission.date.trim() === '' ? false : true;
            if (this.validDate) {
                this.validDateFormat = this.dateValidation();
            }
            if (this.validEmail) {
                this.validEmailFormat = this.emailValidation();
            }
            if (
                this.validName &&
                this.validEmail &&
                this.validFeedback &&
                this.validForm &&
                this.validDate &&
                this.validEmailFormat &&
                this.validDateFormat
            ) {
                if (this.editedSubmissionIndex < 0) {
                    this.$emit('create-new-submission', this.submission);
                } else {
                    this.$emit('edit-the-submission', this.submission, this.editedSubmissionIndex);
                }
            }
        },
        closeForm() {
            this.$emit('close-form');
        },
        dateValidation() {
            var reg = /(0[1-9]|[12][0-9]|3[01])[- /.](0[1-9]|1[012])[- /.](19|20)\d\d/;
            if (this.submission.date.match(reg)) {
                let parts = this.submission.date.split('/');
                var enteredTime = new Date(parts[1] + '/' + parts[0] + '/' + parts[2]);
                var timeNow = new Date();
                if (enteredTime <= timeNow) {
                    return true;
                }
            }
            return false;
        },
        removeNameError() {
            this.validName = true;
        },
        removeDateError() {
            this.validDate = true;
            this.validDateFormat = true;
        },
        removeEmailError() {
            this.validEmail = true;
            this.validEmailFormat = true;
        },
        removeFormError() {
            this.validForm = true;
        },
        removeFeedbackError() {
            this.validFeedback = true;
        },
        emailValidation() {
            const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
            return re.test(String(this.submission.email).toLowerCase());
        }
    }
};
</script>

<style scoped>
textarea {
    resize: none;
}
.es-required-field {
    color: #ff3636;
    display: block;
    font-size: 14px;
    margin-top: 8px;
    text-align: left;
}
</style>
