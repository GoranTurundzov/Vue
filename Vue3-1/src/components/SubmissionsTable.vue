<template>
    <div class="block-content">
        <div class="block">
            <select class="btn btn-default dropdown-toggle" @change="postsPerPage">
                <option value="5" selected>5</option>
                <option value="10">10</option>
                <option value="15">15</option>
                <option value="20">20</option>
            </select>
            <button
                class="btn btn-secondary margin-left"
                @click="toggleFilter"
                data-testid="open-filters-button"
            >
                Filter submissions
            </button>
            <button
                class="btn btn-primary pull-right"
                @click="createNewSubmission"
                data-testid="create-submission-button"
            >
                Create submission
            </button>
        </div>
        <div class="block block-bordered block-rounded">
            <submissions-table-row
                v-for="(item, index) in loadedSubmissions"
                :key="index"
                :row-data="item"
                class="row-hover"
                @activate-modal="triggerModal"
                @delete-submission="deleteSubmission"
                @edit-submission="editSubmission"
            ></submissions-table-row>
        </div>
        <div class="row justify-content-center">
            <button
                class="btn btn-primary justify-content-center"
                data-testid="test-load-more"
                v-if="showButton"
                @click="loadMore"
            >
                Load More
            </button>
        </div>
    </div>
    <modal v-if="modalActive" name="previev-submission-modal" @close-modal="triggerModal">
        <template v-slot:header>
            <h3>{{ modalData.name }}'s Submission</h3>
        </template>
        <submissions-table-row :align-in-rows="true" :row-data="modalData"></submissions-table-row>
    </modal>
    <modal v-if="filterModal" @close-modal="toggleFilter" name="filter-modal">
        <template v-slot:header>
            <h3>Filter Submissions</h3>
        </template>
        <submission-table-filter
            :prop-form-type="appliedFilters.form_type"
            :prop-status="appliedFilters.status"
            :prop-date="appliedFilters.date"
            :prop-recommendation="appliedFilters.recommendation"
            :prop-form="appliedFilters.form"
            :form-select-options="formSelectOptions"
            @cancel-filters="toggleFilter"
            @change-filter="stockFilter"
        >
        </submission-table-filter>
        <template v-slot:footer>
            <button class="btn btn-seconday" @click="toggleFilter">
                Cancel
            </button>
            <button
                class="btn btn-primary"
                @click="submitFilter"
                data-testid="submit-filters-button"
            >
                Continue
            </button>
        </template>
    </modal>
</template>

<script>
import SubmissionsTableRow from '@/components/SubmissionsTableRow';
import Modal from '@/components/Modal';
import SubmissionTableFilter from '@/components/SubmissionTableFilter';
import filterData from '../services/filterData';

export default {
    name: 'SubmissionsTable',
    components: {
        SubmissionsTableRow,
        Modal,
        SubmissionTableFilter
    },
    props: {
        tableData: {
            type: Array,
            required: true,
            description: 'it provides data for each row of the table'
        },
        formSelectOptions: {
            type: Array,
            required: true,
            description: 'Forms to select from'
        }
    },
    emits: ['delete-this-submission', 'create-new-submission', 'edit-this-submission'],
    data() {
        return {
            rowsToLoad: 5,
            timesLoaded: 1,
            filterModal: false,
            modalActive: false,
            modalData: null,
            selectedFilters: null,
            appliedFilters: {
                form: 'All',
                form_type: 'All',
                recommendation: 'All',
                date: 'All',
                status: 'All'
            }
        };
    },
    computed: {
        loadedSubmissions() {
            let loadedData = this.tableData;
            let filteredData = filterData(loadedData, this.appliedFilters);
            return filteredData.slice(0, this.rowsToLoad * this.timesLoaded);
        },
        showButton() {
            return this.rowsToLoad * this.timesLoaded <= this.loadedSubmissions.length;
        }
    },
    methods: {
        loadMore() {
            this.timesLoaded++;
        },
        postsPerPage(e) {
            this.timesLoaded = 1;
            this.rowsToLoad = e.target.value;
        },
        toggleFilter() {
            this.filterModal = !this.filterModal;
        },
        stockFilter(data) {
            this.selectedFilters = data;
        },
        submitFilter() {
            this.appliedFilters.form = this.selectedFilters.form;
            this.appliedFilters.status = this.selectedFilters.status;
            this.appliedFilters.recommendation = this.selectedFilters.recommendation;
            this.appliedFilters.date = this.selectedFilters.date;
            this.appliedFilters.form_type = this.selectedFilters.form_type;
            this.toggleFilter();
        },
        triggerModal(passedData) {
            this.modalData = passedData;
            this.modalActive = !this.modalActive;
        },
        deleteSubmission(item) {
            let deleteIndex = this.tableData.findIndex(x => x == item);
            this.$emit('delete-this-submission', deleteIndex);
        },
        editSubmission(item) {
            this.$emit('edit-this-submission', item);
        },
        createNewSubmission() {
            this.$emit('create-new-submission');
        }
    }
};
</script>

<style>
.margin-left {
    margin-left: 10px;
}
.row-hover:hover {
    box-shadow: 2px 2px 4px 4px rgba(0, 0, 0, 0.2);
    transform: scale(1.001);
}
</style>
