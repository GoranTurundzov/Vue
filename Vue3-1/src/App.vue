<template>
    <div class="app">
        <submissions-table
            v-if="!showSubmissionForm && data && formSelectOptions"
            :table-data="data"
            :form-select-options="formSelectOptions"
            @delete-this-submission="confirmationDelete"
            @create-new-submission="toggleSubmissionForm"
            @edit-this-submission="editSubmission"
        ></submissions-table>
        <submission-form
            v-if="showSubmissionForm && formSelectOptions"
            :edited-submission-index="submissionIndexedForEditing"
            :submission-to-edit="submissionForEditing"
            :formSelectOptions="formSelectOptions"
            @edit-the-submission="editSubmissionSubmit"
            @create-new-submission="createNewSubmission"
            @close-form="closeForm"
        ></submission-form>
        <modal 
            v-if="deleteModal" 
            name="delete-modal" 
            data-testid="delete-modal-component"
            @close-modal="toggleDeleteModal"
        >
            <template v-slot:header>
                <h3>Delete submission</h3>
            </template>
            <p>Are you sure you want to delete this submission?</p>
            <template v-slot:footer>
                <button class="btn btn-secondary" @click="toggleDeleteModal">
                    <i class="fa fa-times"></i> Cancel
                </button>
                <button
                    class="btn btn-primary"
                    @click="deleteSubmission"
                    data-testid="confirm-delete-button"
                >
                    <i class="fa fa-check"></i> Confirm
                </button>
            </template>
        </modal>
    </div>
</template>

<script>
import SubmissionsTable from './components/SubmissionsTable';
import submissionCall from './services/SubmissionJsonService';
import formatter from './services/SubmissionsFormatter';
import Modal from './components/Modal.vue';
import formTypes from './services/formSelectionService';
import SubmissionForm from './components/SubmissionForm.vue';

export default {
    name: 'App',
    components: {
        SubmissionsTable,
        Modal,
        SubmissionForm
    },
    data() {
        return {
            rawData: null,
            data: null,
            deleteModal: false,
            submissionIndexedForDeletion: null,
            formSelectOptions: null,
            showSubmissionForm: false,
            submissionIndexedForEditing: -1,
            submissionForEditing: null
        };
    },
    async created() {
        try {
            let rawData = await submissionCall();
            let formatedData = formatter(rawData);
            this.data = formatedData;
            let formData = await formTypes();
            this.formSelectOptions = formData;
        } catch (error) {
            throw new Error(error);
        }
    },
    methods: {
        deleteSubmission() {
            this.data.splice(this.submissionIndexedForDeletion, 1);
            this.toggleDeleteModal();
        },
        confirmationDelete(index) {
            this.deleteModal = true;
            this.submissionIndexedForDeletion = index;
        },
        toggleDeleteModal() {
            this.deleteModal = !this.deleteModal;
        },
        toggleSubmissionForm() {
            this.showSubmissionForm = !this.showSubmissionForm;
        },
        createNewSubmission(submission) {
            submission.form_type = submission.form.form_type;
            submission.form = submission.form.name;
            this.data.unshift(submission);
            this.showSubmissionForm = false;
        },
        closeForm() {
            this.submissionIndexedForEditing = -1;
            this.submissionForEditing = null;
            this.toggleSubmissionForm();
        },
        editSubmission(item) {
            this.submissionForEditing = item;
            this.submissionIndexedForEditing = this.data.findIndex(x => x == item);
            this.toggleSubmissionForm();
        },
        editSubmissionSubmit(submission, index) {
            submission.form_type = submission.form.form_type;
            submission.form = submission.form.name;
            this.data[index] = submission;
            this.closeForm();
        }
    }
};
</script>

<style>
@import url('../src/assets/Styles/bootstrap.min.css');
@import url('../src/assets/Styles/new_site.min.css');
@import url('../src/assets/Styles/style.css');
</style>
