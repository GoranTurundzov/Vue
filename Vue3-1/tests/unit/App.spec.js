import { mount, flushPromises } from '@vue/test-utils';
import App from '@/App';
import SubmissionsTable from '@/components/SubmissionsTable';
import submissionCall from '../../src/services/SubmissionJsonService';
import formatter from '../../src/services/SubmissionsFormatter';
import formTypes from '../../src/services/formSelectionService';
import json from '../../public/submissions.json';
import formJson from '../../public/forms.json';
import dummyData from '../../src/assets/DummyData';
import SubmissionForm from '../../src/components/SubmissionForm';

let wrapper;

jest.mock('../../src/services/SubmissionsFormatter');
jest.mock('../../src/services/SubmissionJsonService');
jest.mock('../../src/services/formSelectionService');
let deleteSubmissionFn = jest.spyOn(App.methods, 'deleteSubmission');
let confirmationDeleteFn = jest.spyOn(App.methods, 'confirmationDelete');
let toggleSubmissionFormFn = jest.spyOn(App.methods, 'toggleSubmissionForm');
let createNewSubmissionFn = jest.spyOn(App.methods, 'createNewSubmission');
let editSubmissionSubmitFn = jest.spyOn(App.methods, 'editSubmissionSubmit');

describe('App', () => {
	
    beforeEach(async () => {
        jest.clearAllMocks();

        wrapper = mount(App);

        await flushPromises();
    });

    it('App renders', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Calls on the formatter function', () => {
        expect(formatter).toHaveBeenCalledTimes(1);
    });

    it('Calls on the form function', () => {
        expect(formTypes).toHaveBeenCalledTimes(1);
    });

    test('It Renders the SubmissionsTable', async () => {
        await wrapper.setData({
            data: dummyData,
            formSelectOptions: formJson
        });
        expect(wrapper.findComponent(SubmissionsTable).exists()).toBe(true);
    });

    it('Makes a request when the page is created', async () =>
        submissionCall.mockResolvedValueOnce(json));

    it('Uses the formated json data', async () => {
        let formatedData = await formatter();
        expect(wrapper.vm.data).toEqual(formatedData);
    });

    it('Deletes emited component', async () => {
        await wrapper.setData({
            data: dummyData,
            formSelectOptions: formJson
        });
        let table = wrapper.findComponent(SubmissionsTable);
        expect(wrapper.vm.submissionIndexedForDeletion).toBeNull();
        expect(wrapper.vm.data[3].name).toEqual('Nicole');

        await table.vm.$emit('delete-this-submission', 3);

        expect(wrapper.vm.submissionIndexedForDeletion).toBe(3);
        expect(confirmationDeleteFn).toBeCalled();
        expect(wrapper.vm.deleteModal).toBe(true);
        expect(wrapper.find('[data-testid="delete-modal-component"]').exists()).toBe(true);
        expect(wrapper.find('[data-testid="confirm-delete-button"]').exists()).toBe(true);
        
        await wrapper.find('[data-testid="confirm-delete-button"]').trigger('click');
        
        expect(deleteSubmissionFn).toBeCalled();
        //Ivana previously was at index 4 and after the deletion of Nicole she drops to index 3
        expect(wrapper.vm.data[3].name).toEqual('Ivana Trajkova');
    });

    test('It renders the SubmissionForm', async () => {
        await wrapper.setData({
            data: dummyData,
            formSelectOptions: formJson,
            submissionForm: false
        });
        expect(wrapper.findComponent(SubmissionForm).exists()).toBe(false);
        wrapper.find('[data-testid="create-submission-button"]').trigger('click');
        await expect(wrapper.findComponent(SubmissionsTable).emitted('create-new-submission')).toBeTruthy();
        expect(toggleSubmissionFormFn).toBeCalled();
        expect(wrapper.findComponent(SubmissionForm).exists()).toBe(true);
    });

    test('Submits and adds the new submission', async () => {
        await wrapper.setData({
            data: dummyData,
            formSelectOptions: formJson,
            showSubmissionForm: true
        });
        let createdSubmission = {
            name: 'Goran',
            email: 'Goran@goan.com',
            recommendation: 'yes',
            feedback: 'no feedback whatsoever',
            date: '11/11/2020',
            form: {
                name: 'Form 1',
                form_type: 'popup'
            },
            status: 'pending'
        };
        let submissionForm = wrapper.findComponent(SubmissionForm);
        await submissionForm.vm.$emit('create-new-submission', createdSubmission);
        expect(createNewSubmissionFn).toBeCalled();
        expect(wrapper.vm.data[0].name).toEqual('Goran');
        expect(wrapper.vm.data[0].email).toEqual('Goran@goan.com');
        expect(wrapper.vm.data[0].recommendation).toEqual('yes');
        expect(wrapper.vm.data[0].feedback).toEqual('no feedback whatsoever');
        expect(wrapper.vm.data[0].form).toEqual('Form 1');
        expect(wrapper.vm.data[0].form_type).toEqual('popup');
        expect(wrapper.vm.data[0].status).toEqual('pending');
    });

    it('Changes the edited submission', async () => {
        await wrapper.setData({
            data: dummyData,
            formSelectOptions: formJson,
            showSubmissionForm: true
        });
        let editedSubmission = {
            name: 'Goran',
            email: 'Goran@goan.com',
            recommendation: 'yes',
            feedback: 'no feedback whatsoever',
            date: '11/11/2020',
            form: {
                name: 'Form 1',
                form_type: 'popup'
            },
            status: 'pending'
        };
        expect(wrapper.vm.data[2]).toEqual({
            name: 'Ivona Atanasova',
            email: 'ivana_ar@example.com',
            recommendation: 'maybe',
            feedback: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non enim gravida ante feugiat fermentum in vitae sem. Mauris ligula ante, finibus quis conva',       
            date: '10/11/2020',
            form: 'Form 2',
            form_type: 'page',
            status: 'approved'
          });
        let submissionForm = wrapper.findComponent(SubmissionForm);
        await submissionForm.vm.$emit('edit-the-submission', editedSubmission, 2);
        expect(editSubmissionSubmitFn).toBeCalled();
        expect(wrapper.vm.data[2]).toEqual({
            name: 'Goran',
            email: 'Goran@goan.com',
            recommendation: 'yes',
            feedback: 'no feedback whatsoever',
            date: '11/11/2020',
            form: 'Form 1',
            status: 'pending',
            form_type: 'popup'
        });
    });
});
