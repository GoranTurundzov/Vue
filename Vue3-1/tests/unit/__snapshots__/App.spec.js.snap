// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders 1`] = `
<div
  class="app"
>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>
`;
